packages/jest-util/src/convertDescriptorToString.js

```diff
@@ -23,7 +23,7 @@ export default function convertDescriptorToString(
     throw new Error('describe expects a class, function, number, or string.');
   }
 
-  if (descriptor.name !== undefined) {
+  if (descriptor.name != undefined) {
```

Comment:
?

Reply:
`descriptor.name !== undefined`
always evaluates to true

Reply:
```js
const descriptor = {};

console.log(descriptor.name !== undefined); // false
```

Reply:
@SimenB let me know your thoughts

Reply:
I meant comparing dissimilar types using the strict equality operators === and !== will always return the same value, respectively false and true, because no type conversion is done before the comparison. Thus, such comparisons can only be bugs.

`var a = 8;
  var b = "8";
  if (a === b) {  // Noncompliant; always false
    // ...
  }
`

Reply:
We also do `typeof foo !== 'undefined'` which is "better" because it handles undeclared variables

<img width="424" alt="screenshot 2019-01-21 17 36 43" src="https://user-images.githubusercontent.com/2440089/51490489-52f25a00-1da3-11e9-97f7-c9afd6c99d52.png">